src/GenericApp.js

```diff
@@ -304,9 +362,13 @@ class GenericApp extends Router {
         });
     }
 
+    /**
+     * Gets the IP addresses of the given host.
+     * @param {string} host
+     */
     getIpAddresses(host) {
         return new Promise((resolve, reject) => {
-            this.socket.socket.emit('getHostByIp', host || this.common.host, (ip, _host) => {
+            this.socket._socket.emit('getHostByIp', host || this.common.host, (ip, _host) => {
```

Comment:
@GermanBluefox this is the kind of bugs TypeScript helps avoid. Since you are already using gulp to compile the sources and you are using PropTypes (which is a bit like TypeScript but only at runtime), I think the step to static typechecking is not that large.

If adapter-react is going to be the basis for all react-based adapters (which definitely makes sense), we need to ensure the highest possible quality.
I strongly believe that switching to TypeScript would help immensely in doing so, especially because it makes it clear what every parameter and variable looks like without guessing or having to go look at the source code of methods. If you like, we can discuss this further somewhere else.

Reply:
If we decide to use TypeScript, you can also auto-generate the PropTypes from the type declarations - so no need to update them in two locations.